client/state/sites/reducer.js

```diff
@@ -103,8 +103,6 @@ export function items( state = {}, action ) {
 			}, initialNextState );
```

Comment:
The thinking with the condition is that we'd only previously tracked the currently selected site in state. We'd want to allow updates to be received and applied for those selected site(s), but not all sites until all sites are contained within state. Otherwise we'd be duplicating data between sites list and state, and much of the state data would go unused (since "all sites" data continued to consume exclusively from sites-list).

With recent changes to sites-list, it may be the case that we're now tracking all sites in state? If so, the initial condition could/should be dropped altogether.